openedx/features/course_experience/views/course_home_messages.py

```diff
@@ -124,19 +127,26 @@ def _register_course_home_messages(request, course, user_access, course_start_da
             ),
             title=Text(_('You must be enrolled in the course to see course content.'))
         )
-    if not user_access['is_anonymous'] and not user_access['is_staff'] and not user_access['is_enrolled']:
-        CourseHomeMessages.register_info_message(
-            request,
-            Text(_(
-                u'{open_enroll_link}Enroll now{close_enroll_link} to access the full course.'
-            )).format(
-                open_enroll_link=HTML('<button class="enroll-btn btn-link">'),
-                close_enroll_link=HTML('</button>')
-            ),
-            title=Text(_(u'Welcome to {course_display_name}')).format(
-                course_display_name=course.display_name
+    if not user_access['is_anonymous'] and not user_access['is_staff'] and \
+            not user_access['is_enrolled']:
+        if not course.invitation_only:
+            CourseHomeMessages.register_info_message(
+                request,
+                Text(_(
+                    u'{open_enroll_link}Enroll now{close_enroll_link} to access the full course.'
+                )).format(
+                    open_enroll_link=HTML('<button class="enroll-btn btn-link">'),
+                    close_enroll_link=HTML('</button>')
+                ),
+                title=Text(_(u'Welcome to {course_display_name}')).format(
+                    course_display_name=course.display_name
+                )
+            )
+        else:
+            CourseHomeMessages.register_info_message(
+                request,
+                Text(_(u'You should be enrolled in the course to fully access the contents.')),
```

Comment:
Can we use the [`DEFAULT_PUBLIC_VIEW_MESSAGE`](https://github.com/edx/edx-platform/blob/4aa64465334f5c6c04282f435160b4e0ec8a12ae/common/lib/xmodule/xmodule/x_module.py#L75-L78) that was originally added by this feature?